components/asset.rst

```diff
@@ -0,0 +1,251 @@
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Applications that host their assets on different domains and CDNs (*Content
+Delivery Networks*) should use instead the ``UrlPackage`` class to generate
+absolute URLs for their assets::
+
+    use Symfony\Component\Asset\UrlPackage;
+
+    $package = new UrlPackage('http://static.example.com/images/', new StaticVersionStrategy('v1'));
+
+    echo $package->getUrl('/logo.png');
+    // result: http://static.example.com/images/logo.png?v1
+
+In case you serve assets from more than one domain to improve application
+performance, pass an array of URLs as the first argument of ``UrlPackage``
+constructor::
+
+    use Symfony\Component\Asset\UrlPackage;
+
+    $urls = array(
+        'http://static1.example.com/images/',
+        'http://static2.example.com/images/',
+    );
+    $package = new UrlPackage($urls, new StaticVersionStrategy('v1'));
+
+    echo $package->getUrl('/logo.png');
+    // result: http://static1.example.com/images/logo.png?v1
+
+The selection of the domain which will serve the asset is deterministic, meaning
+that each asset will be always served by the same domain. This behavior simplifies
+the management of HTTP cache.
+
+Request Context Aware Assets
+............................
+
+Similarly to application-relative assets, absolute assets can also take into
+account the context of the current request. In this case, only the request
+scheme is considered, in order to select the appropriate base URL (HTTPs or
+protocol-relative URLs for HTTPs requests, any base URL for HTTP requests)::
+
+    use Symfony\Component\Asset\UrlPackage;
+    use Symfony\Component\Asset\Context\RequestStackContext;
+
+    $package = new UrlPackage(array('http://example.com/', 'https://example.com/'), new StaticVersionStrategy('v1'));
+    $package->setContext(new RequestStackContext($requestStack));
+
+    echo $package->getUrl('/logo.png');
+    // result: https://example.com/logo.png?v1
+
+Named Packages
+~~~~~~~~~~~~~~
+
+Applications that manage lots of different assets may need to group them in
+packages with the same versioning strategy and base path. The Asset component
+includes a ``Packages`` class to simplify the management of several packages.
+
+In the following example, all packages use the same versioning strategy, but
+they all have different base paths::
+
+    use Symfony\Component\Asset\Package;
+    use Symfony\Component\Asset\PathPackage;
+    use Symfony\Component\Asset\UrlPackage;
+    use Symfony\Component\Asset\Packages;
+
+    $versionStrategy = new StaticVersionStrategy('v1');
+
+    $defaultPackage = new Package($versionStrategy);
+
+    $namedPackages = array(
+        'img' => new UrlPackage('http://img.example.com/', $versionStrategy),
+        'doc' => new PathPackage('/somewhere/deep/for/documents', $versionStrategy),
+    );
+
+    $packages = new Packages($defaultPackage, $namedPackages)
+
+The ``Packages`` class requires to define a default package which will be applied
```

Comment:
it doesn't require it. It's only requried as soon as you don't always use named packages